app/com/gu/floodgate/DynamoDBTable.scala

```diff
@@ -71,6 +84,20 @@ trait DynamoDBTable[T] extends StrictLogging {
     getListOfItems(request)
   }
 
+  def getLatestItem(id: String, filterKeyName: String, filterValue: String)(implicit ec: ExecutionContext): Future[Option[T]] = {
+    val request = new QueryRequest().withTableName(tableName)
+      .withScanIndexForward(false)
+      .withKeyConditionExpression(s"$keyName = :$keyName")
```

Comment:
Yep. In `.withKeyConditionExpression(s"$keyName = :$keyName")`, the second instance is a placeholder to which you bind a value a few lines below with `.withExpressionAttributeValues(Map(s":$keyName" -> new AttributeValue().withS(id)))`

The name of the placeholder doesn't matter, as long as it doesn't conflict with other placeholders in the same expression. So you could just write:

`.withKeyConditionExpression(s"$keyName = :keyName")`

and

`.withExpressionAttributeValues(Map(s":keyName" -> new AttributeValue().withS(id)))`